tests/qasm3/declarations/test_classical.py

```diff
@@ -603,6 +612,43 @@ def test_duration_casting_error(qasm_code, error_message, error_span, caplog):
     assert error_span in caplog.text
 
 
+@pytest.mark.parametrize(
```

Comment:
We should also add the tests which validate the assignment of the expressions which contain `pi` and their equivalent converted bit string rep. Essentially we want to make sure that the bit representation is reflective of the conversion method we are using